pkg/clusteragent/externalmetrics/model/datadogmetricinternal.go

```diff
@@ -154,6 +157,14 @@ func (d *DatadogMetricInternal) UpdateFrom(current datadoghq.DatadogMetric) {
 	if d.shouldResolveQuery(currentSpec) {
 		d.resolveQuery(currentSpec.Query)
 	}
+
+	if d.query != currentSpec.Query ||
```

Comment:
could you please add a comment that explains why we reset the Retry info here. 🙇

Reply:
Done!